src/integrationTest/kotlin/com/pubnub/api/integration/PublishIntegrationTests.kt

```diff
@@ -135,6 +137,49 @@ class PublishIntegrationTests : BaseIntegrationTest() {
         success.listen()
     }
 
+    @Test
+    fun testEventListenerApi() {
+
+        val success = AtomicBoolean()
+        val expectedChannel = randomChannel()
+        val messagePayload = generateMessage(pubnub)
+
+        val observer = createPubNub()
+
+        pubnub.addListener(object : SubscribeCallback() {
+            override fun status(pubnub: PubNub, pnStatus: PNStatus) {
+                if (pnStatus.operation == PNOperationType.PNSubscribeOperation &&
+                    pnStatus.affectedChannels.contains(expectedChannel)
+                ) {
+                    observer.publish(
+                        message = messagePayload,
+                        channel = expectedChannel
+                    ).async { _, status ->
+                        assertFalse(status.error)
+                    }
+                }
+            }
+        })
+
+        val testChannel = pubnub.channel(expectedChannel)
+        val testSubscription = testChannel.subscription(ChannelOptions.receivePresenceEvents())
+        testSubscription.addListener(object : EventListener() {
```

Comment:
Is there a way using current design to do following:


        val testChannel = pubnub.channel(expectedChannel)
        val testSubscription = testChannel.subscription(ChannelOptions.receivePresenceEvents())
        val myOnMessageForSub01: (String) -> Unit = { println("-=Handling MESSAGE for one channel: Channel01. Message: $it") }

         **testSubscription.onMessage = myOnMessageForSub01**

?

This case has been presented in POC created by Michał some time ago:
https://docs.google.com/document/d/1lsebtJl84JbNsgkmNpeND7ie3nU0ebN7HSvvMXVbeP0/edit#heading=h.td0xl84g2462

Reply:
not yet, but it's possible to build it on top of the current design, which I'm proposing to do in a future PR